p2p/switch.go

```diff
@@ -907,25 +908,72 @@ func (sw *Switch) addPeer(p Peer) error {
 			continue
 		}
 
-		chs := reactor.GetChannels()
-		if len(chs) > 0 { // for each reactor there is exactly one PeerSet, no need to iterate over channels
-			if err := sw.peerSetByChID[chs[0].ID].Add(p); err != nil {
+		peerSet := sw.peerSetForReactor(reactor)
+		if peerSet != nil {
+			if err := peerSet.Add(p); err != nil {
 				if errors.Is(err, ErrPeerRemoval{}) {
-					sw.Logger.Error("Error starting peer ",
-						" err ", "Peer has already errored and removal was attempted.",
+					sw.Logger.Error("Error starting peer",
+						"err", "Peer has already errored and removal was attempted",
 						"peer", p.ID())
 				}
-				return err // TODO(tzdybal): this should never happen, right?
+				return err
 			}
 			peerWanted = true
 		}
 	}
 	if !peerWanted {
 		sw.Logger.Error("Peer not wanted by any reactor", "peer", p)
-		sw.StopPeerGracefully(p)
+		sw.StopPeerGracefully(p, "")
 	}
 
 	sw.Logger.Debug("Added peer", "peer", p)
 
 	return nil
 }
+
+// peerSetForReactor retrieves the PeerSet associated with the first channel of the given Reactor. Returns nil if empty.
+// For each reactor there is exactly one PeerSet, no need to iterate over channels
+func (sw *Switch) peerSetForReactor(r Reactor) *PeerSet {
+	if len(r.GetChannels()) > 0 {
+		return sw.peerSetByChID[r.GetChannels()[0].ID]
+	}
+	return nil
+}
+
+// removePeerFromAllReactors removes the given peer from all reactors
+func (sw *Switch) removePeerFromAllReactors(peer Peer, reason interface{}) {
+	for _, reactor := range sw.reactors {
+		sw.doRemovePeer(peer, reactor, reason)
+	}
+}
+
+// removePeerFromReactor removes the peer from the specified reactor
+func (sw *Switch) removePeerFromReactor(peer Peer, reactorName string) {
+	for _, reactor := range sw.reactors {
+		if reactor.String() == reactorName {
+			sw.doRemovePeer(peer, reactor, nil)
+			break
+		}
+	}
+}
+
+// doRemovePeer removes the specified peer from the given reactor and its corresponding peer set, logging any issues.
+func (sw *Switch) doRemovePeer(peer Peer, reactor Reactor, reason interface{}) {
```

Comment:
why the `do` prefix in the function name?

Reply:
Good catch, this can be skipped. I think I planned to make it removePeerFromReactor (which is a conflicting name), and so added `do` prefix, and then removed `FromReactor` suffix :dizzy_face: